ETHCHAT.md

```diff
@@ -0,0 +1,230 @@
+---
+slug: 70
+title: 70/ETH-SECPM
+name: Private 1:1 messages over Ethereum
+status: raw
+category: Standards Track
+tags: wake-application
+editor: Ramses Fernandez <ramses@status.im>
+contributors:
+---
+
+# Abstract
+This document specifies an Ethereum-based private messaging service. 
+This proposal is built upon the model [https://rfc.vac.dev/spec/20/](https://rfc.vac.dev/spec/20/) and 
+amends the limitations of the latter concerning forward privacy and authentication.
+
+# Background
+
+Alice wants to send an encrypted message to Bob. 
+Here Bob is the only individual able to decrypt the message.
+Alice has access to Bob’s Ethereum address.
+
+# Theory and Description of the Protocol
+
+The proposed protocol must adhere to the following design requirements:
+-   Alice knows Bob’s Ethereum address.
+-   Bob is willing to participate in the protocol, and publishes his public key.
+-   Bob’s ownership of his public key is verifiable,
+-   Alice wants to send message M to Bob.
+-   An eavesdropper cannot read M’s content even if she is storing it or relaying it.
+
+The specification is based on the noise protocol framework.
+It corresponds to the double ratchet scheme combined with the X3DH algorithm, which will be used to initialize the former. 
+We chose to express the protocol in noise to be be able to use the noise streamlined implementation and proving features.
+The X3DH algorithm provides both authentication and forward secrecy, as stated in the [X3DH specification](https://signal.org/docs/specifications/x3dh/).
+
+## High level description
+This protocol will consist of several stages:
+
+1.  Key setting for X3DH: this step will produce prekey bundles for Bob which will be fed into X3DH. It will also allow Alice to generate the keys required to run the X3DH algorithm correctly.
+2.  Execution of X3DH: This step will output a common secret key SK together with an additional data vector AD. Both will be used in the Double Ratchet algorithm initialization.
+3.  Execution of the Double Ratchet algorithm for forward secure, authenticated communications, using the common secret key SK, obtained from X3DH, as a root key.
+
+## Cryptographic functions required
+-   XEdDSA for digital signatures involved in the X3DH key generation.
+-   SHA256 for hashing and the generation of HMACs.
+-   AES256-CBC for the encryption/decryption of messages.
+
```

Comment:
Are these algorithms strict, or is it possible to choose one like a cipher suite like in TLS?